src/libpriv/rpmostree-core.c

```diff
@@ -789,6 +791,33 @@ rpmostree_context_setup (RpmOstreeContext    *self,
   if (g_variant_dict_contains (self->spec->dict, "ignore-scripts"))
     sd_journal_print (LOG_INFO, "ignore-scripts is no longer supported");
 
+  gboolean selinux;
+  g_assert (g_variant_dict_lookup (self->spec->dict, "selinux", "b", &selinux));
+  /* Load policy from / if SELinux is enabled, and we haven't already loaded
+   * a policy.  This is mostly for the "compose tree" case.
+   */
+  if (selinux)
+    {
+      glnx_autofd int host_rootfs_dfd = -1;
+      /* Ensure that the imported packages are labeled with *a* policy if
+       * possible, even if it's not the final one. This helps avoid duplicating
+       * all of the content.
+       */
+      if (!glnx_opendirat (AT_FDCWD, "/", TRUE, &host_rootfs_dfd, error))
+        return FALSE;
+      g_autoptr(OstreeSePolicy) sepolicy = ostree_sepolicy_new_at (host_rootfs_dfd, cancellable, error);
+      if (!sepolicy)
+        return FALSE;
+      /* For system contexts, whether SELinux required is dynamic based on the
+       * filesystem state. If the base compose doesn't have a policy package,
+       * then that's OK. It'd be cleaner if we loaded the no-SELinux state from
+       * the origin, but that'd be a backwards-incompatible change.
+       */
+      if (!self->is_system && ostree_sepolicy_get_name (sepolicy) == NULL)
+        return glnx_throw (error, "Unable to load SELinux policy from /");
+      rpmostree_context_set_sepolicy (self, sepolicy);
```

Comment:
Though... this now overwrites the explicit `set_sepolicy` we did before calling `setup` in the upgrader path, right? That means less potential sharing between the cache and the final commit. I.e. maybe just:

```diff
-if (selinux)
+if (selinux && !self->sepolicy)
```

?